diff --git a/node_modules/@langchain/core/dist/language_models/chat_models.cjs b/node_modules/@langchain/core/dist/language_models/chat_models.cjs
index 81ee732..8531fdd 100644
--- a/node_modules/@langchain/core/dist/language_models/chat_models.cjs
+++ b/node_modules/@langchain/core/dist/language_models/chat_models.cjs
@@ -157,6 +157,19 @@ class BaseChatModel extends base_js_1.BaseLanguageModel {
                 await Promise.all((runManagers ?? []).map((runManager) => runManager?.handleLLMError(err)));
                 throw err;
             }
+            // --- START: MANUAL PATCH: AGENT TOKEN DOUBLING FIX---
+            if (
+              generationChunk?.message?.usage_metadata &&
+              llmOutput?.tokenUsage
+            ) {
+              generationChunk.message.usage_metadata.input_tokens =
+                llmOutput.tokenUsage.promptTokens;
+              generationChunk.message.usage_metadata.output_tokens =
+                llmOutput.tokenUsage.completionTokens;
+              generationChunk.message.usage_metadata.total_tokens =
+                llmOutput.tokenUsage.totalTokens;
+            }
+            // --- END: MANUAL PATCH: AGENT TOKEN DOUBLING FIX---
             await Promise.all((runManagers ?? []).map((runManager) => runManager?.handleLLMEnd({
                 // TODO: Remove cast after figuring out inheritance
                 generations: [[generationChunk]],
